src/animalShelter/AnimalShelter.ts

```diff
@@ -68,8 +69,33 @@ export default class AnimalShelter {
       console.log(
         `Cannot change reptile limit to ${newReptileLimit} there ${reptilesTotal == 1 ? "is" : "are"} already ${reptilesTotal} ${reptilesTotal == 1 ? "reptile" : "reptile"}`
       );
+    };
+  };
+
+  private hasMethod(animal:Animal, methodName:string): boolean{
+    return typeof (animal as any)[methodName] === "function";
```

Comment:
you shouldn't cast to any

Reply:
Usually yes, but in this case, in my opinion, we are sure it is an animal since it has been passed as one

Reply:
@Kamen06  still not a reason to cast to "any"

Reply:
i cannot think of another way of checking if an object has a specific function

Reply:
I still think we shouldn't be using this `hasMethod`, it's a weird way of reporting the animal's abilities. @rkoychev isn't it better to add properties `canJump` and `canRun` the same way we have for climbing the trees?